test/sample_set_filter_test.py

```diff
@@ -0,0 +1,222 @@
+                         set(['ww163e']))
+
+    def test_filter_number_eq(self):
+        samples_eq = self.filter([self.sample_set_1_ref], [{
+            'column': "longitude",
+            'comparison': "==",
+            'value': "-92.1833",
+            'condition': "AND",
+        }])
+
+        self.assertEqual(len(samples_eq), 1)
+        self.assertEqual(set(sample['name'] for sample in samples_eq),
+                         set(['Core 1-1*-1M']))
+
+    def test_merge_sets_string_eq(self):
+        samples_merge = self.filter([self.sample_set_1_ref, self.sample_set_2_ref], [{
+            'column': "sesar:collection_method",
+            'comparison': "==",
+            'value': "Coring > Syringe",
+            'condition': "AND",
+        }])
+        self.assertEqual(len(samples_merge), 207)
+
+    def test_str_in(self):
+        samples_merge = self.filter([self.sample_set_2_ref], [{
+            'column': "sesar:collection_method",
+            'comparison': "in",
+            'value': "Coring > Syringe, Grab",
+            'condition': "AND",
+        }])
+        self.assertEqual(len(samples_merge), 209)
+
+    def test_and(self):
+        samples_merge = self.filter([self.sample_set_2_ref], [
+            {
+                'column': "sesar:collection_method",
+                'comparison': "==",
+                'value': "Coring > Syringe",
+                'condition': "AND",
+            },
+            {
+                'column': "latitude",
+                'comparison': ">",
+                'value': "33.33",
+                'condition': "OR",
+            }
+        ])
+        self.assertEqual(len(samples_merge), 160)
+
+    def test_or(self):
+        samples_merge = self.filter([self.sample_set_2_ref], [
+            {
+                'column': "sesar:collection_method",
+                'comparison': "==",
+                'value': "Coring > Syringe",
+                'condition': "OR",
+            },
+            {
+                'column': "latitude",
+                'comparison': ">",
+                'value': "33.33",
+                'condition': "AND",
+            }
+        ])
+        self.assertEqual(len(samples_merge), 210)
+
+    def test_order_of_ops(self):
+        samples_merge = self.filter([self.sample_set_2_ref], [
+            {
+                'column': "sesar:collection_method",
+                'comparison': "==",
+                'value': "Coring > Syringe",
+                'condition': "OR",
+            },
+            {
+                'column': "latitude",
+                'comparison': ">",
+                'value': "33.33",
+                'condition': "AND",
+            },
+            {
+                'column': "longitude",
+                'comparison': "<",
+                'value': "81.718",
+                'condition': "OR",
+            }
+        ])
+        self.assertEqual(len(samples_merge), 206)
```

Comment:
Are there any inputs where you would expect to see an error / errors? What if the sample ref doesn't exist, or the condition is weird?

Reply:
No conditions should error. Otherwise I would expect this sort of testing to be handled by the sample search API tests (it is iirc).